tests/test_pytorch_geometric.py

```diff
@@ -0,0 +1,113 @@
+    def test_scatter(self):
+        if torch.cuda.is_available():
+            return
+        src = torch.tensor([[2, 0, 1, 4, 3], [0, 2, 1, 3, 4]])
+        index = torch.tensor([[4, 5, 4, 2, 3], [0, 0, 2, 2, 1]])
+        out, argmax = torch_scatter.scatter_max(src, index, fill_value=0)
+        test_argmax = torch.LongTensor([[-1, -1,  3,  4,  0,  1],
+                                        [ 1,  4,  3, -1, -1, -1]])
+        test_out = torch.LongTensor([[0, 0, 4, 3, 2, 0],
+                                     [2, 4, 3, 0, 0, 0]])
+        self.assertTrue(torch.all(torch.eq(test_argmax, argmax)))
+        self.assertTrue(torch.all(torch.eq(test_out, out)))
+
+    def test_cluster(self):
+        if torch.cuda.is_available():
+            return
+        x = torch.Tensor([[-1, -1], [-1, 1], [1, -1], [1, 1]])
+        batch = torch.tensor([0, 0, 0, 0])
+        edge_index = torch_cluster.knn_graph(x, k=2, batch=batch, loop=False)
+        test_edge_index = torch.LongTensor([[2, 1, 3, 0, 3, 0, 1, 2],
+                                            [0, 0, 1, 1, 2, 2, 3, 3]])
+        self.assertTrue(torch.all(torch.eq(test_edge_index, edge_index)))
+
+    def test_sparse(self):
+        if torch.cuda.is_available():
+            return
+        index = torch.tensor([[1, 0, 1, 0, 2, 1],
+                              [0, 1, 1, 1, 0, 0]])
+        value = torch.Tensor([[1, 2], [2, 3], [3, 4], [4, 5], [5, 6], [6, 7]])
+
+        index, value = torch_sparse.transpose(index, value, 3, 2)
+        test_index = torch.LongTensor([[0, 0, 1, 1],
+                                       [1, 2, 0, 1]])
+        test_value = torch.Tensor([[7., 9.], [5., 6.], [6., 8.], [3., 4.]])
+        self.assertTrue(torch.all(torch.eq(test_index, index)))
+        self.assertTrue(torch.all(torch.eq(test_value, value)))
+
+    def test_spline_conv(self):
+        if torch.cuda.is_available():
+            return
+        in_channels, out_channels = (16, 32)
+        edge_index = torch.tensor([[0, 0, 0, 1, 2, 3], [1, 2, 3, 0, 0, 0]])
+        num_nodes = edge_index.max().item() + 1
+        x = torch.randn((num_nodes, in_channels))
+        pseudo = torch.rand((edge_index.size(1), 3))
+
+        conv = SplineConv(in_channels, out_channels, dim=3, kernel_size=5)
+        assert conv.__repr__() == 'SplineConv(16, 32)'
+        with torch_geometric.debug():
+           assert conv(x, edge_index, pseudo).size() == (num_nodes, out_channels)
+
+    def test_geometric(self):
+        if torch.cuda.is_available():
+            return
+        x = torch.tensor([[1, 3, 5], [2, 4, 6]], dtype=torch.float).t()
+        edge_index = torch.tensor([[0, 0, 1, 1, 2], [1, 1, 0, 2, 1]])
+        data = Data(x=x, edge_index=edge_index).to(torch.device('cpu'))
+
+        N = data.num_nodes
+
+        self.assertTrue(data.x.tolist() == x.tolist())
+        self.assertTrue(data['x'].tolist() == x.tolist())
+
+        self.assertTrue(sorted(data.keys) == ['edge_index', 'x'])
```

Comment:
Use self.assertCountEqual(...) to compare to list when you don't care about the ordering:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual

Applies throughout to all list comparison.